uui-components/src/table/DataTableTimelineHeaderCell.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { DataTableHeaderCellProps, IHasCX } from '@epam/uui-core';
+import { FlexCell } from '../layout';
+
+/**
+ * Header cell props of DataTable with Timeline.
+ */
+export interface DataTableTimelineHeaderCellProps<TItem, TId> extends DataTableHeaderCellProps<TItem, TId>, IHasCX {
+    /**
+     * Renders custom header cell for timeline.
+     * @returns Header cell content.
+     */
+    renderCellContent: () => React.ReactNode;
+}
+
+export function DataTableTimelineHeaderCell<TItem, TId>(props: DataTableTimelineHeaderCellProps<TItem, TId>) {
```

Comment:
The other name for DataTableTimelineHeaderCell should be discussed. Maybe, that is an overhead for such a case.
The name @jakobz suggested is `DataTableCustomHeaderCell` or something similar.